src/taipy/core/_backup/_backup.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+
+import os
+import pathlib
+
+__BACKUP_FILE_PATH_ENVIRONMENT_VARIABLE_NAME = "TAIPY_BACKUP_FILE_PATH"
+
+
+def append_to_backup_file(new_file_path: str):
+    if preserve_file_path := os.getenv(__BACKUP_FILE_PATH_ENVIRONMENT_VARIABLE_NAME, None):
+        with open(preserve_file_path, "a") as f:
+            f.write(f"{new_file_path}\n")
+
+
+def remove_from_backup_file(to_remove_file_path: str):
+    preserve_file_path = os.getenv(__BACKUP_FILE_PATH_ENVIRONMENT_VARIABLE_NAME, None)
+    if preserve_file_path and pathlib.Path(preserve_file_path).is_file():
+        with open(preserve_file_path, "r+") as f:
+            lines = f.readlines()
+            to_remove_file_path = f"{to_remove_file_path}\n"
+            if to_remove_file_path in lines:
+                f.seek(0)
+                f.truncate()
+                lines.remove(to_remove_file_path)
+                f.writelines(lines)
```

Comment:
after discussion with @jrobinAV 
There is a problem with:
```
to_remove_file_path = to_remove_file_path + "\n"
new_backup = old_backup.replace(to_remove_file_path, '')
```
in case of nested files:
```
foo/data_node.csv
data_node.csv
bar/data_node.csv
```
So a better solution can be:
```
to_remove_file_path = to_remove_file_path + "\n"
if old.backup.startwith(to_remove_file_path):
    new_backup = old_backup.replace(to_remove_file_path, '', 1)
# Others
new_backup = old_backup.replace("\n" + to_remove_file_path, '')
```

Reply:
if there's a duplication, do we want to remove both dups? Because I can create 2 data nodes that have access to the same file, if I remove 1 data node, it means I'm removing all same file path, and the data node that still exists will be affected as well